assets/js/base/components/totals/totals-shipping-item/has-shipping-rate.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Searches an array of packages/rates to see if there are actually any rates
+ * available.
+ *
+ * @param {Array} shippingRatePackages An array of packages and rates.
+ * @return {boolean} True if a rate exists.
+ */
+const hasShippingRate = ( shippingRatePackages ) => {
+	let foundRate = false;
+
+	shippingRatePackages.forEach( ( shippingRatePackage ) => {
+		if ( shippingRatePackage.shipping_rates.length ) {
+			foundRate = true;
+		}
+	} );
+
+	return foundRate;
```

Comment:
this could be an `Array.some`

```suggestion
	return shippingRatePackages.some( shippingRatePackage  => shippingRatePackage.shipping_rates.length );
```

fancier

```js
	return shippingRatePackages.some( ( { shipping_rates } )  => shipping_rates.length );
```

Reply:
Agreed, and `some` will also abort as soon as something is truthy so _slightly_ more performant (although with the number of rates involved, the difference would be miniscule).